FactoryMethod/src/main/java/com/ben/abstractnotification/FactoryProvider.java

```diff
@@ -0,0 +1,15 @@
+package com.ben.abstractnotification;
+
+public class FactoryProvider {
+    public static AbstractFactory getFactory(String choice){
+
+        if("Phone".equalsIgnoreCase(choice)){
+            return new PhoneFactory();
+        }
+        else if("Notification".equalsIgnoreCase(choice)){
+            return new NotificationFactory();
+        }
+
+        return null;
```

Comment:
What happens if I call `FactoryProvider.getFactory("NotAFactory");`? It might be better here to throw some kind of error, otherwise the client might face a `NullPointerException` when they try and use a factory that doesn't exist

Reply:
Good observation